backend/src/realtime/realtime-note/realtime-note.service.spec.ts

```diff
@@ -89,15 +106,76 @@ describe('RealtimeNoteService', () => {
     setIntervalSpy = jest.spyOn(global, 'setInterval');
     clearIntervalSpy = jest.spyOn(global, 'clearInterval');
 
+    clientWithReadWrite = new MockConnectionBuilder(realtimeNote)
+      .withAcceptingRealtimeUserStatus()
+      .withLoggedInUser(readWriteUsername)
+      .build();
+
+    clientWithRead = new MockConnectionBuilder(realtimeNote)
+      .withDecliningRealtimeUserStatus()
+      .withLoggedInUser(onlyReadUsername)
+      .build();
+
+    clientWithoutReadWrite = new MockConnectionBuilder(realtimeNote)
+      .withDecliningRealtimeUserStatus()
+      .withGuestUser(noAccessUsername)
+      .build();
+
+    clientWithoutReadWrite.getTransporter();
+    clientWithReadWrite.getTransporter();
+
     realtimeNoteService = new RealtimeNoteService(
       revisionsService,
       consoleLoggerService,
       realtimeNoteStore,
       schedulerRegistry,
       mockedAppConfig,
+      mockedPermissionService,
     );
   });
 
+  it('should not remove the connection with read and write access', async () => {
+    jest.spyOn(realtimeNoteStore, 'find').mockImplementation(() => {
+      return realtimeNote;
+    });
+
+    const loggedUserTransporter = clientWithReadWrite.getTransporter();
+
+    jest.spyOn(loggedUserTransporter, 'disconnect');
+
+    await realtimeNoteService.handleNotePermissionChanged(note);
+
+    expect(loggedUserTransporter.disconnect).toHaveBeenCalledTimes(0);
+  });
+
+  it('should close the connection for removed connection', async () => {
+    jest.spyOn(realtimeNoteStore, 'find').mockImplementation(() => {
+      return realtimeNote;
+    });
+    const guestUserTransporter = clientWithoutReadWrite.getTransporter();
+    jest.spyOn(guestUserTransporter, 'disconnect');
+
+    await realtimeNoteService.handleNotePermissionChanged(note);
+
+    expect(guestUserTransporter.disconnect).toHaveBeenCalledTimes(1);
+  });
+  describe('acceptEdit', () => {
+    beforeEach(() => {
+      jest.spyOn(realtimeNoteStore, 'find').mockImplementation(() => {
+        return realtimeNote;
+      });
+    });
```

Comment:
if you're doing this then put the other two tests that test `handleNotePermissionChanged` into the describe block as well. Especially because they're doing the same spy call.

Reply:
Please remember to rename the describe block then.

Reply:
renamed the block and moved tests to same describe block